compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -266,6 +277,14 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     }
 }
 
+/// Formats a number of bytes into a human readable SI-prefixed size.
+/// Returns a tuple of `(quantity, units)`.
+pub fn human_readable_bytes(bytes: u64) -> (u64, &'static str) {
+    static UNITS: [&str; 7] = ["B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB"];
+    let i = ((bytes.ilog2() / 10) as usize).min(UNITS.len() - 1);
```

Comment:
`0u64.ilog2()` panics with "argument of integer logarithm must be positive". Currently that's not really an issue for how we use this function but I think it would be best if we went ahead and made it work for all possible values of `u64` in case this function is ever re-used in a context where `bytes` could be zero.

Reply:
Using `checked_ilog2().unwrap_or(0)` instead